src/main/java/org/hydev/mcpm/client/database/searchusecase/SearcherByCommand.java

```diff
@@ -0,0 +1,40 @@
+package org.hydev.mcpm.client.database.searchusecase;
+
+import org.hydev.mcpm.client.models.PluginModel;
+
+import java.util.*;
+
+/**
+ * Searcher that returns a map based on commands.
+ *
+ * @author Jerry Zhu (<a href="https://github.com/jerryzhu509">...</a>)
+ */
+public class SearcherByCommand implements Searcher {
+
+    private static Map<String, List<PluginModel>> commandMap = null;
+
+    /**
+     * Returns a dictionary mapping the different commands to the matching plugins.
+     *
+     * @param plugins A list of all plugins in the database.
+     * @return A dictionary associating the command to the matching plugins.
+     */
+    @Override
+    public Map<String, List<PluginModel>> constructSearchMaps(List<PluginModel> plugins) {
+        //TODO
+        return new HashMap<>();
+    }
+
+    /**
+     * Searches for plugins based on the provided user input.
+     *
+     * @param inp User input for the search. Should be a name as a string here.
+     * @param plugins A list of all plugins in the database.
+     * @return A dictionary associating a string feature of the plugins to the matching plugins.
+     */
+    @Override
+    public List<PluginModel> getSearchList(Object inp, List<PluginModel> plugins) {
```

Comment:
This is a default method, but I think a default behaviour for searching makes too much sense. You could leave this abstract instead, so the compiler will remind you to override the search method (which is a core functionality).

Reply:
I disagree unless I'm misunderstanding what you're saying... Are you saying we want our Searcher interface to provide a default behaviour for this method?

Reply:
I am saying we do not. I missed a word

Reply:
Oh good. How about `throw NotImplementedException()`?